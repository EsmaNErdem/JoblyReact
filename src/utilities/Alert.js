const Alert = ({ type, messages }) => {

}

export default Alert;