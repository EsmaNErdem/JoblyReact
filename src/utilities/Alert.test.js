import { render } from "@testing-library/react";
import Alert from "./Alert";

test("renders without crashing", async () => {
    render(<Alert />); 
});

test("it renders and matches with snaphot", () => {
    const { asFragment } = render(<Alert type="success" messages={["Test-success"]} />);
    expect(asFragment()).toMatchSnapshot();
});

test("it renders and matches with snaphot", () => {
    const { asFragment } = render(<Alert type="danger" messages={["Test-danger"]} />);
    expect(asFragment()).toMatchSnapshot();
});


test("it renders alert with success class", () => {
    const { getByText } = render(<Alert type="success" messages={["Test-success"]} />);
    const alertElement = getByText("Test-success").parentElement;
    expect(alertElement).toHaveClass("Alert-success");
});

test("it renders alert with danger class", () => {
    const { getByText } = render(<Alert type="danger" messages={["Test-danger"]} />);
    const alertElement = getByText("Test-danger").parentElement;
    expect(alertElement).toHaveClass("Alert-danger");
});

