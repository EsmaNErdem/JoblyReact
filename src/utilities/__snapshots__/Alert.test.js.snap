// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders and matches with snaphot 1`] = `
<DocumentFragment>
  <div
    class="Alert, Alert-success"
  >
    <p>
      Test-success
    </p>
  </div>
</DocumentFragment>
`;

exports[`it renders and matches with snaphot 2`] = `
<DocumentFragment>
  <div
    class="Alert, Alert-danger"
  >
    <p>
      Test-danger
    </p>
  </div>
</DocumentFragment>
`;

exports[`it renders and matches with snaphot 3`] = `
<DocumentFragment>
  <div
    class="Alert, Alert-danger"
  >
    <p>
      hello
    </p>
  </div>
</DocumentFragment>
`;
